networkx/algorithms/tree/coding.py

```diff
@@ -312,7 +312,7 @@ def parents(u):
         if v < index and degree[v] == 1:
             u = v
         else:
-            index = u = min(k for k in range(index + 1, n) if degree[k] == 1)
+            index = u = next(k for k in range(index + 1, n) if degree[k] == 1)
```

Comment:
next instead correctly chooses the first satisfying k, avoiding unnecessarily scanning the remainder of the range. Since the index which is shifted forwards, defines the start of the next range, the entire range from 0 to index is only scanned once. Therefore the running time is linear.